pytealext/state.py

```diff
@@ -124,6 +123,22 @@ def __getitem__(self, index: Union[int, Expr]):
         raise NotImplementedError
 
 
+class StateArray2D(StateArray):
+    """
+    Wrapper for state access which utilizes multiple slots organized in 2D array
+    """
+    def __init__(self, prefix: Union[str, Expr]):
+        """
+        Args:
+            prefix: a key prefix in the global state, if it's a string it will be converted to Bytes
+        """
+        if isinstance(prefix, str):
+            prefix += "-"
+        else:
+            prefix = Concat(prefix, Bytes("-"))
```

Comment:
I suggest that we don't do this and let the users handle the prefixes on their own - maybe add it as a suggestion to class docstring

Reply:
The key won't be readable either way :)

Reply:
I was thinking here about names' conflicts. If there is no addition, I think it might be a bit easier to bump into naming conflicts. Do you think adding some description to how prefixes in 2D work will help?